### PR TITLE
[#2774] feat(doc): Add docs for messaging catalog

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

This PR proposes to add docs for messaging catalog

### Why are the changes needed?

Fix: #2774 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

No

### CODE REVIEW

File: docs/kafka-catalog.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+slug: /kafka-catalog
+date: 2024-4-22
+keyword: kafka catalog
+license: "Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2."
+---
+
+## Introduction
+
+Kafka catalog is a messaging catalog that offers the ability to manage Kafka topics metadata.
```
Comment (jerryshao) [inline docs/kafka-catalog.md:12]: You'd better change to Apache Kafka for the first time you mentioned about Kafka.

File: docs/manage-messaging-metadata-using-gravitino.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+TopicCatalog topicCatalog = catalog.asTopicCatalog();
+NameIdentifier[] identifiers =
+    topicCatalog.listTopics(Namespace.ofTopic("metalake", "catalog", "default"));
+// ...
+```
+
+</TabItem>
+</Tabs>
+
```
Comment (jerryshao) [inline docs/manage-messaging-metadata-using-gravitino.md:308]: Can you please remove the trailing blank lines.

File: docs/manage-messaging-metadata-using-gravitino.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+### Alter a schema
+
+Please refer to [Alter a schema](./manage-relational-metadata-using-gravitino.md#alter-a-schema)
+in relational catalog for more details. For a messaging catalog, the alter operation is the same.
+
+### Drop a schema
+
+Please refer to [Drop a schema](./manage-relational-metadata-using-gravitino.md#drop-a-schema)
+in relational catalog for more details. For a messaging catalog, the drop operation is the same.
```
Comment (yuqi1129) [inline docs/manage-messaging-metadata-using-gravitino.md:132]: Is the scheme in the Kafka catalog read-only?

Comment (mchades) [reply docs/manage-messaging-metadata-using-gravitino.md:132]: yes. see https://github.com/datastrato/gravitino/pull/3106/files#diff-656563ad28ab24f645bb63b4974d71f2d7c115a3547aacb44ea679fa75758ec3R34

**Pull request merged**